lib/rubocop/cop/factory_bot/factory_association_with_strategy.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module FactoryBot
+      # Use definition in factory association instead of hard coding a strategy.
+      #
+      # @example
+      #   # bad - only works for one strategy
+      #   factory :foo do
+      #     profile { create(:profile) }
+      #   end
+      #
+      #   # good - implicit
+      #   factory :foo do
+      #     profile
+      #   end
+      #
+      #   # good - explicit
+      #   factory :foo do
+      #     association :profile
+      #   end
+      #
+      #   # good - inline
+      #   factory :foo do
+      #     profile { association :profile }
+      #   end
+      #
+      class FactoryAssociationWithStrategy < ::RuboCop::Cop::Base
+        MSG = 'Use an implicit, explicit or inline definition instead of ' \
+              'hard coding a strategy for setting association within factory.'
+
+        HARDCODED = Set.new(%i[create build build_stubbed]).freeze
+
+        # @!method factory_declaration(node)
+        def_node_matcher :factory_declaration, <<~PATTERN
+          (block (send nil? {:factory :trait} ...)
+            ...
+          )
+        PATTERN
+
+        # @!method factory_strategy_association(node)
+        def_node_matcher :factory_strategy_association, <<~PATTERN
+          (block
+            (send nil? _association_name)
+            (args)
+            < $(send nil? HARDCODED ...) ... >
+          )
+        PATTERN
+
+        def on_block(node)
+          factory_declaration(node) do
+            node.each_node do |statement|
+              factory_strategy_association(statement) do |hardcoded_association|
+                add_offense(hardcoded_association)
+              end
+            end
+          end
+        end
+        alias on_numblock on_block
```

Comment:
Do we really need to handle the numblock case? `factory`/`trait` don't yield any arguments, do they?
I'd disable the internal affairs cop that suggest that, or would skip adding a spec to avoid the added confusion gerarding `factory` block arguments.

Reply:
Indeed it is. Let's suppress the offense caused by removing this 👍Thanks for catching that!

Reply:
Done.